packages/synapse-react-client/src/utils/functions/SqlFunctions.ts

```diff
@@ -27,93 +28,112 @@ export const getIgnoredQueryFilterSearchParamKey = (
   return `__${namespace ?? ''}_${key}`
 }
 /**
- * Given the search params, return a set of QueryFilters to narrow the the query to view just related data. May return null if a QueryFilter should not be added.
+ * Look in local storage for a set of QueryFilters to apply.  In addition, given the search params,
+ * generate a set of QueryFilters to narrow the the query to view just related data.
+ * May return null if a QueryFilter should not be added.
  * @param sql
  * @param searchParams
  * @param operator
  * @returns
  */
-export const generateQueryFilterFromSearchParams = (
+export const getAdditionalFilters = (
   searchParams?: Record<string, string>,
   operator: SQLOperator = ColumnSingleValueFilterOperator.LIKE,
 ): QueryFilter[] | undefined => {
-  if (!searchParams) {
-    return
-  }
-  const isQueryWrapperKey = (key: string) =>
-    key.startsWith('QueryWrapper') || key.startsWith('__')
-  const searchParamKeys = Object.keys(searchParams)
-  if (
-    searchParamKeys.length === 0 ||
-    searchParamKeys.every(isQueryWrapperKey)
-  ) {
-    return
+  const localStorageQueryFiltersString = localStorage.getItem(
+    QUERY_FILTERS_LOCAL_STORAGE_KEY,
+  )
+  let additionalFilters: QueryFilter[] = []
+  if (localStorageQueryFiltersString) {
+    additionalFilters = JSON.parse(
+      localStorageQueryFiltersString,
+    ) as QueryFilter[]
+    // delay clearing out the value so re-rendering during React Strict mode does not destroy the filter
+    setTimeout(
+      () => localStorage.removeItem(QUERY_FILTERS_LOCAL_STORAGE_KEY),
+      1000,
+    )
   }
+  if (searchParams) {
+    const isQueryWrapperKey = (key: string) =>
+      key.startsWith('QueryWrapper') || key.startsWith('__')
+    const searchParamKeys = Object.keys(searchParams)
+    if (
+      searchParamKeys.length === 0 ||
+      searchParamKeys.every(isQueryWrapperKey)
+    ) {
+      return
```

Comment:
Bug here where filters from localStorage will not be returned if we have a searchParams object with no query filter keys.

You could return `additionalFilters.length === 0 ? undefined : additionalFilters` here, but see next comment for suggested fix